scipy/optimize/_linprog_rs.py

```diff
@@ -0,0 +1,364 @@
+        b = B.b                     # similar to b[ab[i][l]] = j
+    else:
+        status = 1
+
+    return x, b, status, iteration
+
+
+# FIXME: is maxiter for each phase?
+def _linprog_rs(c, c0, A, b, callback=None, maxiter=1000, tol=1e-12,
+                maxupdate=10, mast=False, pivot="mrc", **unknown_options):
+    """
+    Solve the following linear programming problem via a two-phase
+    revised simplex algorithm.::
+
+        minimize:     c^T @ x
+
+        subject to:  A @ x == b
+                     0 <= x < oo
+
+    Parameters
+    ----------
+    c : 1D array
+        Coefficients of the linear objective function to be minimized.
+    c0 : float
+        Constant term in objective function due to fixed (and eliminated)
+        variables. (Currently unused.)
+    A : 2D array
+        2D array which, when matrix-multiplied by ``x``, gives the values of
+        the equality constraints at ``x``.
+    b : 1D array
+        1D array of values representing the RHS of each equality constraint
+        (row) in ``A_eq``.
+    callback : callable, optional (Currently unused.)
+
+    Options
+    -------
+    maxiter : int
+       The maximum number of iterations to perform in either phase.
+    tol : float
+        The tolerance which determines when a solution is "close enough" to
+        zero in Phase 1 to be considered a basic feasible solution or close
+        enough to positive to serve as an optimal solution.
+    maxupdate : int
+        The maximum number of updates performed on the LU factorization.
+        After this many updates is reached, the basis matrix is factorized
+        from scratch.
+    mast : bool
+        Minimize Amortized Solve Time. If enabled, the average time to solve
+        a linear system using the basis factorization is measured. Typically,
+        the average solve time will decrease with each successive
+        solve after initial factorization, as factorization (and updates) take
+        less time than the solve operation. Eventually, however, the updated
+        factorization becomes sufficiently complex that the average solve time
+        begins to increase. When this is detected, the basis is refactorized
+        from scratch. Enable this option to maximize speed at the risk of
+        nondeterministic behavior. Ignored if ``maxupdate`` is 0.
+    pivot : "mrc" or "bland"
+        Pivot rule: Minimum Reduced Cost (default) or Bland's rule. Choose
+        Bland's rule if iteration limit is reached and cycling is suspected.
+
+    Returns
+    -------
+    x : 1D array
+        Solution vector.
+    status : int
+        An integer representing the exit status of the optimization::
+
+         0 : Optimization terminated successfully
+         1 : Iteration limit reached
+         2 : Problem appears to be infeasible
+         3 : Problem appears to be unbounded
+         4 : Numerical difficulties encountered
+         5 : No constraints; turn presolve on
```

Comment:
small thing: status ``5`` should also be defined in the top level ``linprog(...)`` function.

Reply:
Maybe. Then again, it won't ever be returned by the other solvers, and it's a pathological case for a user to include no constraints AND disable presolve. I only created this status to pass test `test_unbounded_below_no_presolve_corrected` so maybe I should just change that test instead of adding this silly status?

Reply:
I've found another [comment](https://github.com/scipy/scipy/blob/a6009217b66f58b7b4e445ff365686f48845e33a/scipy/optimize/_linprog_util.py#L1170) of status 5; are these supposed to be related?